### DISCUSSION TITLE
How to implement contact backup to cloud (export to VCF)?

### DISCUSSION THREAD

**Discussion by rohaitas** [vestrel00/contacts-android#323] [Q&A] [answered]
I am developing an app where I have to backup contacts to cloud storage. For that, I suppose I need a VCF file. But how to get the VCF file so that I can upload it to cloud storage?  I can only get vcf file with share intent. Is there another way?

`  Intent shareIntent = ContactShareKt.shareVCardIntent(contacts);
            if (shareIntent != null) {
               v.getContext().startActivity(Intent.createChooser(shareIntent, null));
            }`

**Comment by vestrel00** [comment, marked as answer]
> I can only get vcf file with share intent. Is there another way?

Currently, there is no other way with this library. 

However, read/write from/to VCF is in the project roadmap with an issue already created for it; #26

It is currently not a high priority feature at the moment for me as there are other more important issues to tackle first (e.g. adding unit tests). Furthermore, it will require quite a bit of time and effort to implement properly.

I will most likely use https://github.com/mangstadt/ez-vcard internally to implement #26

In the meantime, you can still use the query APIs provided in this library to read contact data easily, and then use https://github.com/mangstadt/ez-vcard to export data that you want into a VCF file.

Let's keep this discussion open though. If this gets enough upvotes, then I may move this up the priority queue 😉

**Reply by vestrel00** [reply]
Marking this as the answer for now. Thanks for the question!
